File: java_backend/src/main/java/com/example/dto/Node.java
package com.example.dto;

import java.util.Date;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Node {
	private String id;
	private String map_id;
	private String title;
	private String node_type;
	private int duration;
	private String description;
	private int progress_rate;
	private String parent_id;
	private List<String> children_ids;
	private Date created_at;
	private Date updated_at;
	private Date due_at;
	private Date finished_at;
	
}

File: flutter_client/lib/model/entities/node.dart
import 'package:flutter/material.dart';

enum NodeType { start, goal, normal, root }

class Node {
  int id;
  int? parentId;
  List<int> childrenIds;
  NodeType nodeType;
  String title;
  String description;
  int duration;
  int progressRate;
  double x;
  double y;
  DateTime dueAt;
  DateTime? finishedAt;
  DateTime createdAt;
  DateTime updatedAt;

  Node({
    required this.id,
    required this.parentId,
    required this.childrenIds,
    required this.nodeType,
    required this.title,
    required this.description,
    required this.duration,
    required this.progressRate,
    required this.x,
    required this.y,
    required this.dueAt,
    this.finishedAt,
    required this.createdAt,
    required this.updatedAt,
  });
}
